compiler/rustc_hir_typeck/src/method/confirm.rs

```diff
@@ -516,11 +516,9 @@ impl<'a, 'tcx> ConfirmContext<'a, 'tcx> {
                         .report_mismatched_types(&cause, method_self_ty, self_ty, terr)
                         .emit();
                 } else {
-                    span_bug!(
+                    self.dcx().span_delayed_bug(
```

Comment:
what specifically is this for?

having hit this when refactoring method autoderef stuff, reasonably worried this will cause an ice later lol

Reply:
https://github.com/rust-lang/rust/blob/master/tests/ui/self/arbitrary-self-from-method-substs.rs

will hit that in `default` mode on the method call in `main` (during typeck).

I guess I could try to see if I can poison whatever emits the error in `get` and then check that in typeck.

Reply:
I changed it to an `error!` plus `has_errors().unwrap()`, meaning we won't ever silently eat it, and if there were no errors we'll escalate to an ICE.